File: MyBudgetSpring/src/main/java/com/example/mybudgetspring/model/dto/Transaction.java
package com.example.mybudgetspring.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transaction implements Serializable {
    private Integer transactionId;
    private String description;
    private Double amount;
    private Account account;
}

File: MyBudgetAngular/mybudgetapp/src/app/model/transaction.ts
import { Account } from './account';

export class Transaction {
  transactionId: number;
  description: string;
  amount: number;
  defaultCurrencyAmount: number;
  account: Account;

  constructor(
    transactionId: number,
    description: string,
    amount: number,
    defaultCurrencyAmount: number,
    account: Account
  ) {
    this.transactionId = transactionId;
    this.description = description;
    this.amount = amount;
    this.defaultCurrencyAmount = defaultCurrencyAmount;
    this.account = account;
  }
}
